lms/djangoapps/course_home_api/progress/views.py

```diff
@@ -204,6 +204,9 @@ def get(self, request, *args, **kwargs):
         collected_block_structure = get_block_structure_manager(course_key).get_collected()
         course_grade = CourseGradeFactory().read(student, collected_block_structure=collected_block_structure)
 
+        # recalculate course grade from visible grades (stored grade was calculated over all grades, visible or not)
+        course_grade.update(visible_grades_only=True, has_staff_access=is_staff)
```

Comment:
How does this play with masquerading users?

Reply:
It should maintain parity with the legacy view: staff (masquerading or not) can see all of the target user's (or their own) grades.

You'll see that we pass in the student user in the line above this and the `is_staff` check is for the requesting user (not the student). So visibility will be determined by the requesting user's staff status.

It feels more natural to me to only ignore visibility when actually masquerading (like, what if a staff user was taking a course themselves?) but that's not what legacy does and in general, we don't cater too much to staff being restricted in any way.